test/runtests.jl

```diff
@@ -16,36 +16,48 @@ A = [ 1.7     0     0     0     0     0     0     0   .13     0
 A = sparse(A)
 
 for perm ∈ (collect(1 : A.n), amd(A))
-  L, d, α, P = lldl(A, perm, memory=0)
```

Comment:
Want to add METIS.jl to the list of orderings?

Reply:
Discussion avec Mathieu Tanneau : "METIS met plus longtemps à calculer l’ordonnancement, mais j’ai eu des factorisations jusqu’à 5x plus rapides avec. Le speed up était là avec Suitespares et avec LDLFactorization."

Reply:
```
julia> include("example.jl")
AMD ordering
  p  nnz(K)  nnz(L)  ‖LDLᵀ-K‖        α      time
  0    8677    8677  1.52e-01  0.0e+00  1.54e-03
  5    8677   12962  5.17e-02  0.0e+00  1.96e-03
 10    8677   17051  2.91e-02  0.0e+00  2.27e-03
 15    8677   20932  1.27e-02  0.0e+00  2.70e-03
 20    8677   24466  6.27e-03  0.0e+00  2.91e-03
 25    8677   27816  3.40e-03  0.0e+00  3.19e-03
 30    8677   30984  1.89e-03  0.0e+00  3.49e-03
 35    8677   33889  1.08e-03  0.0e+00  3.81e-03
 40    8677   36569  6.05e-04  0.0e+00  4.16e-03
 45    8677   39123  3.32e-04  0.0e+00  4.35e-03
 50    8677   41552  1.80e-04  0.0e+00  4.77e-03
 55    8677   43767  9.57e-05  0.0e+00  4.85e-03
 60    8677   45808  5.20e-05  0.0e+00  5.07e-03
 65    8677   47661  2.87e-05  0.0e+00  5.73e-03
 70    8677   49352  1.49e-05  0.0e+00  5.63e-03
 75    8677   50868  7.72e-06  0.0e+00  5.64e-03
 80    8677   52262  3.71e-06  0.0e+00  6.15e-03
 85    8677   53585  1.62e-06  0.0e+00  6.18e-03
 90    8677   54760  7.07e-07  0.0e+00  6.85e-03
 95    8677   55668  3.26e-07  0.0e+00  7.04e-03
100    8677   56283  1.50e-07  0.0e+00  6.25e-03
105    8677   56721  5.57e-08  0.0e+00  6.28e-03
110    8677   57006  1.32e-08  0.0e+00  6.44e-03
115    8677   57177  1.05e-09  0.0e+00  6.54e-03
120    8677   57260  8.82e-13  0.0e+00  6.62e-03
125    8677   57275  2.82e-16  0.0e+00  6.89e-03
130    8677   57275  2.82e-16  0.0e+00  7.91e-03

Metis ordering
  p  nnz(K)  nnz(L)  ‖LDLᵀ-K‖        α      time
  0    8677    8677  8.92e-02  0.0e+00  1.69e-03
  5    8677   13061  4.03e-02  0.0e+00  2.04e-03
 10    8677   17131  2.23e-02  0.0e+00  2.66e-03
 15    8677   20963  1.17e-02  0.0e+00  2.73e-03
 20    8677   24608  6.16e-03  0.0e+00  3.30e-03
 25    8677   28047  3.22e-03  0.0e+00  3.32e-03
 30    8677   31229  1.79e-03  0.0e+00  3.60e-03
 35    8677   34109  9.13e-04  0.0e+00  3.70e-03
 40    8677   36728  4.64e-04  0.0e+00  4.08e-03
 45    8677   39035  2.35e-04  0.0e+00  4.29e-03
 50    8677   40984  1.20e-04  0.0e+00  6.11e-03
 55    8677   42599  5.71e-05  0.0e+00  4.54e-03
 60    8677   43900  2.60e-05  0.0e+00  4.67e-03
 65    8677   44962  1.09e-05  0.0e+00  5.24e-03
 70    8677   45783  3.50e-06  0.0e+00  5.21e-03
 75    8677   46388  6.26e-07  0.0e+00  5.25e-03
 80    8677   46681  3.40e-08  0.0e+00  5.34e-03
 85    8677   46774  1.61e-10  0.0e+00  4.80e-03
 90    8677   46777  2.34e-16  0.0e+00  4.84e-03
 95    8677   46777  2.34e-16  0.0e+00  5.05e-03
100    8677   46777  2.34e-16  0.0e+00  4.94e-03
105    8677   46777  2.34e-16  0.0e+00  4.81e-03
110    8677   46777  2.34e-16  0.0e+00  4.84e-03
115    8677   46777  2.34e-16  0.0e+00  5.60e-03
120    8677   46777  2.34e-16  0.0e+00  4.88e-03
125    8677   46777  2.34e-16  0.0e+00  4.81e-03
130    8677   46777  2.34e-16  0.0e+00  5.49e-03
```

Reply:
I suggest we add METIS.jl to the tests, example and README.

Reply:
Here **only** :-)

Reply:
`nnz(LLDL)` should return the number of nnz of the L factor only or nnz(L) + length(D) ?

Reply:
I would say `nnz(L) + length(D)` because that's how many values we're really storing.

Reply:
With Metis, the permutation returned is an array of `Int32`. With`A` a `SparseMatrixCSC{Tf, Ti}` we have an error if `Ti != Int32`. We should remove some restrictions on type here and in `LDLFactorization` to allow a permutation of Int32 / Int64 whatever the type Ti.